arrays/arrays.c

```diff
@@ -41,25 +41,46 @@ Array *create_array (int capacity) {
 void destroy_array(Array *arr) {
 
   // Free all elements
-  free(arr->elements);
+  for(int x = 0; x < arr->count; x++){ 
+    arr->elements[x] = NULL;
+    free(arr->elements[x]);
+  }
+
 
   // Free array
-free(arr);
+  free(arr->elements);  
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
 void resize_array(Array *arr) {
+  
+
 
   // Create a new element storage with double capacity
 
+  char **newcap = calloc((2 * arr->capacity), sizeof(char *));
+
   // Copy elements into the new storage
 
+  for(int x = 0; x < arr->count; x++) {
+    newcap[x] = arr->elements[x];
+  }
+
+
   // Free the old elements array (but NOT the strings they point to)
```

Comment:
This for loop appears to be unnecessary as it frees the strings the old elements points to.